lib/msal-node/src/client/ConfidentialClientApplication.ts

```diff
@@ -136,8 +138,25 @@ export class ConfidentialClientApplication
             );
         }
 
+        /*
+         * if this env variable is set, and the developer provided region isn't defined and isn't "DisableMsalForceRegion",
+         * MSAL shall opt-in to ESTS-R with the value of this variable
+         */
+        const ENV_MSAL_FORCE_REGION: AzureRegion | undefined =
+            process.env[MSAL_FORCE_REGION];
+
+        let region: AzureRegion | undefined;
+        if (validRequest.azureRegion !== "DisableMsalForceRegion") {
+            region =
+                !validRequest.azureRegion && ENV_MSAL_FORCE_REGION
+                    ? ENV_MSAL_FORCE_REGION
+                    : validRequest.azureRegion;
+        } else {
+            validRequest.azureRegion = undefined;
```

Comment:
Although the test cases are valid and currently passing, the implementation pattern here still looks weird:

```typescript
let sentinel | undefined;

if (condition) {sentinel = something;} else {variable_BAR = another_thing;}

use(sentinel);
```

I suppose this branch could/should be adjusted like this?

```suggestion
            region = undefined;
```

Reply:
oops. good catch, thanks